### PR TITLE
chore: Update cicd.yml file on flame_3d to match main, rollback color changes

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

Update cicd.yml file on flame_3d to match main.

Since we finally updated main, we should require no difference whatsoever on this file anymore.

This will require downgrading all the color changes, which is fine. This should never have been a part of flame_3d to begin with.

Files were "untouched" by checking out the exact version as they are in main right now (will need to rebase flame_3d to main later). I had to add a couple more files because the files on main had dependencies on changes that are not yet rebased on flame_3d. These extra diffs should disappear when I do the final rebase.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org/
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: doc/flame/examples/lib/ray_cast.dart [hunk 1]
```diff
@@ -13,7 +13,7 @@ class RayCastExample extends FlameGame with HasCollisionDetection {
   final velocity = 60;
   double get resetPosition => -canvasSize.y;
 
-  Paint paint = BasicPalette.red.withOpacity(0.6).paint();
+  Paint paint = Paint()..color = Colors.red.withOpacity(0.6);
```
Comment (spydon) [inline doc/flame/examples/lib/ray_cast.dart:16]:
```suggestion
  final Paint paint = Paint()..color = Colors.red.withOpacity(0.6);
```

File: doc/flame/examples/lib/ray_trace.dart [hunk 2]
```diff
@@ -10,7 +10,7 @@ import 'package:flutter/material.dart';
 
 class RayTraceExample extends FlameGame
     with HasCollisionDetection, TapDetector {
-  Paint paint = BasicPalette.red.withOpacity(0.6).paint();
+  Paint paint = Paint()..color = Colors.red.withOpacity(0.6);
```
Comment (spydon) [inline doc/flame/examples/lib/ray_trace.dart:13]:
```suggestion
  final Paint paint = Paint()..color = Colors.red.withOpacity(0.6);
```

File: examples/lib/stories/system/resize_example.dart [hunk 3]
```diff
@@ -13,7 +13,7 @@ class ResizingRectangle extends RectangleComponent {
   void onGameResize(Vector2 size) {
     super.onGameResize(size);
 
-    this.size = size * 0.4;
+    this.size = size * .4;
```
Comment (spydon) [inline examples/lib/stories/system/resize_example.dart:16]:
```suggestion
    this.size = size * 0.4;
```

File: packages/flame/lib/src/components/mixins/has_paint.dart [hunk 4]
```diff
@@ -83,15 +83,12 @@ mixin HasPaint<T extends Object> on Component
       throw ArgumentError('Opacity needs to be between 0 and 1');
     }
 
-    setColor(
-      getPaint(paintId).color.withValues(alpha: opacity),
-      paintId: paintId,
-    );
+    setColor(getPaint(paintId).color.withOpacity(opacity), paintId: paintId);
```
Comment (spydon) [inline packages/flame/lib/src/components/mixins/has_paint.dart:86]: This was easier to read with the trailing comma

File: packages/flame_3d/lib/src/extensions/color.dart [hunk 5]
```diff
@@ -3,5 +3,10 @@ import 'dart:ui';
 
 extension ColorExtension on Color {
   /// Returns a Float32List that represents the color as a vector.
-  Float32List get storage => Float32List.fromList([a, r, g, b]);
+  Float32List get storage => Float32List.fromList([
+        opacity,
+        red.toDouble() / 255,
+        green.toDouble() / 255,
+        blue.toDouble() / 255,
+      ]);
```
Comment (spydon) [inline packages/flame_3d/lib/src/extensions/color.dart:11]: How come this changed? :thinking:

File: packages/flame_tiled/test/tiled_test.dart [hunk 6]
```diff
@@ -182,7 +182,6 @@ void main() {
     test(
```
Comment (spydon) [inline packages/flame_tiled/test/tiled_test.dart:182]: Only the `skips` should be changed in this file right?

File: packages/flame_tiled/test/tile_atlas_test.dart [hunk 7]
```diff
@@ -92,6 +92,7 @@ void main() {
         expect(atlas.key, 'images/green.png');
```
Comment (spydon) [inline packages/flame_tiled/test/tile_atlas_test.dart:92]: Only the `skips` should be changed in this file right?

**Review by spydon — COMMENTED** [review]
Some comments aren't related to this PR

File: doc/flame/examples/lib/ray_cast.dart (hunk 1 above)
Comment (luanpotter) [reply doc/flame/examples/lib/ray_cast.dart:16]: this is copied as is from main - you can fix on a separate PR!

File: doc/flame/examples/lib/ray_trace.dart (hunk 2 above)
Comment (luanpotter) [reply doc/flame/examples/lib/ray_trace.dart:13]: this is copied as is from main - you can fix on a separate PR!

File: examples/lib/stories/system/resize_example.dart (hunk 3 above)
Comment (luanpotter) [reply examples/lib/stories/system/resize_example.dart:16]: this is copied as is from main - you can fix on a separate PR!

File: packages/flame/lib/src/components/mixins/has_paint.dart (hunk 4 above)
Comment (luanpotter) [reply packages/flame/lib/src/components/mixins/has_paint.dart:86]: this is copied as is from main - you can fix on a separate PR!

File: packages/flame_3d/lib/src/extensions/color.dart (hunk 5 above)
Comment (luanpotter) [reply packages/flame_3d/lib/src/extensions/color.dart:11]: I am reverting the color changes to the old version

**Review by spydon — APPROVED** [review]

**Pull request merged**